### PR TITLE
add r4 endpoint resource

### PR DESCRIPTION
**Pull request by eclendenning-va**

### CODE REVIEW

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Endpoint.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  @Valid List<Resource> contained;
+
+  @Valid List<Extension> extension;
+
+  @Valid List<Extension> modifierExtension;
+
+  @Valid List<Identifier> identifier;
+
+  @NotNull Status status;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Endpoint.java:68]: Nit: The spec calls it `EndpointStatus` which I think is more explicit for use in the rest of the code.

File: us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Endpoint.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  @Valid Reference managingOrganization;
+
+  @Valid List<ContactPoint> contact;
+
+  @Valid Period period;
+
+  @Valid @NotEmpty List<CodeableConcept> payloadType;
+
+  @Pattern(regexp = Fhir.CODE)
+  List<String> payloadMimeType;
```
Comment (jhulbert-va) [inline us-core-r4/src/main/java/gov/va/api/health/r4/api/resources/Endpoint.java:83]: Should be:
```
List<@Pattern(regexp = Fhir.CODE) String> payloadMimeType;
```

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/EndpointTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import gov.va.api.health.r4.api.samples.SampleEndpoints;
+import java.util.Collections;
+import org.junit.jupiter.api.Test;
+
+public class EndpointTest {
+
+  private final SampleEndpoints data = SampleEndpoints.get();
+
+  @Test
+  public void Endpoint() {
```
Comment (jhulbert-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/resources/EndpointTest.java:15]: `endpoint` for uniformity.

File: us-core-r4/src/test/java/gov/va/api/health/r4/api/samples/SampleEndpoints.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+@NoArgsConstructor(staticName = "get")
+public class SampleEndpoints {
+  @Delegate SampleDataTypes dataTypes = SampleDataTypes.get();
+
+  public Endpoint endpoint() {
+    return Endpoint.builder()
+        .status(Status.active)
+        .name("My Name")
+        .connectionType(Coding.builder().build())
```
Comment (jhulbert-va) [inline us-core-r4/src/test/java/gov/va/api/health/r4/api/samples/SampleEndpoints.java:17]: There should be sample elements or some other sample class that you can use to fill this.

**Review by jhulbert-va — DISMISSED** [review]

**Review by bschofield-va — APPROVED** [review]

**Pull request merged**